python/ray/air/util/data_batch_conversion.py

```diff
@@ -109,6 +109,47 @@ def convert_pandas_to_batch_type(
         )
 
 
+def _convert_batch_type_to_numpy(
+    data: DataBatchType,
+) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+    """Convert the provided data to a NumPy ndarray or dict of ndarrays.
+
+    Args:
+        data: Data of type DataBatchType
+
+    Returns:
+        A numpy representation of the input data.
+    """
+    if isinstance(data, np.ndarray):
+        return data
+    elif isinstance(data, dict):
+        if data.keys() == {TENSOR_COLUMN_NAME} and isinstance(
+            data[TENSOR_COLUMN_NAME], np.ndarray
+        ):
+            return data[TENSOR_COLUMN_NAME]
+        else:
+            return data
+    elif pyarrow is not None and isinstance(data, pyarrow.Table):
+        if data.column_names == [TENSOR_COLUMN_NAME] and (
+            pyarrow.types.is_floating(data.schema.types[0])
+            or pyarrow.types.is_integer(data.schema.types[0])
```

Comment:
Shouldn't this check for our tensor extension type here, or is this an effort to support "scalar" tensor datasets, where each tensor record is actually a scalar? I don't think that we need support for the latter, if so, e.g. I don't think that any of our Datasets read APIs support constructing such a tensor dataset. 🤔 
```suggestion
            isinstance(data.schema.types[0], TensorDtype)
```

Reply:
I checked this TensorDtype against our existing and new tests cases and realized none of the tests actually hit this path ... am i missing something regarding what `TensorDtype` should capture ?